### PR TITLE
Upgrade to Yarn 4

### PR DESCRIPTION
**Pull request by aryaemami59**
**This PR**:

  - [X] Bumps Node versions in CI to 20
  - [X] Upgrades package manager to Yarn 4
  - [X] Updates `actions/checkout` usages to v4

### CODE REVIEW

File: .codesandbox/ci.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sandboxes": ["vanilla", "vanilla-ts"],
-  "node": "14",
+  "node": "18",
```
Comment (timdorr) [inline .codesandbox/ci.json:3]: This is probably outside the scope of this PR, but do we still need Codesandbox in the CI setup? It seems like unnecessary complexity.

Comment (aryaemami59) [reply .codesandbox/ci.json:3]: I have no idea, I would be happy to get rid of it for you if you in a separate PR if you decide it shouldn't stay.

Comment (phryneas) [reply .codesandbox/ci.json:3]: It gives us immediate releases to play around on every release, and people get sandboxes where those releases are already preinstalled.

Comment (phryneas) [reply .codesandbox/ci.json:3]: For this PR, that would be 
grafik

Unfortunately, it's a bit hidden:
grafik

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -82,7 +82,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: ['18.x']
+        node: ['20.x']
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (aryaemami59) [reply .github/workflows/test.yml:88]: Sure thing!

**Review by timdorr — APPROVED** [review]
Got rid of those warnings, except for the compressed-size-action that doesn't have an upgrade path yet. LGTM!

**Comment by aryaemami59** [comment]
@timdorr Were the warnings caused by `actions/setup-node`?

**Comment by timdorr** [comment]
That and the artifact actions. I got everything I could.

**Comment by aryaemami59** [comment]
@timdorr Thank you so much!

**Comment by timdorr** [comment]
Thanks!

**Pull request merged**